src/utilities/createRouterRoutes.ts

```diff
@@ -10,13 +10,17 @@ type ParentContext = {
   parentDepth?: number,
 }
 
-export function createRouterRoutes(routes: Routes, parentContext: ParentContext = {}): RouterRoute[] {
+export function createRouterRoutes<T extends Routes>(routes: T, parentContext: ParentContext = {}): RouterRoute[] {
   const { parentPath = [], parentQuery = [], parentMatches = [], parentDepth = 0 } = { ...parentContext }
 
   return routes.reduce<RouterRoute[]>((value, route) => {
     const path = typeof route.path === 'string' ? createPath(route.path, {}) : route.path
     const query = typeof route.query === 'string' ? createQuery(route.query, {}) : route.query ?? { query: '', params: {} }
 
+    if (hasDuplicateParams(path, query)) {
+      throw new DuplicateParamsError()
```

Comment:
We probably want to error with the name of the param that was duplicated and where it was (in the path vs the query).